### ISSUE TITLE
Warn if split chunk is not supported.

### ISSUE THREAD

**Issue by yisar** [evanw/esbuild#1581]
Imagine the following code:

```js
esbuild.build({
  entryPoints: {
      a: './a.js',
      b:'./b.js'
    }
})

// c.js
export const ref = {}

// a.js
import {ref} from './c'
ref.value = 'hello'

//b.js
import {ref} from './c'
console.log(ref) // undefined
```

When building multiple pages, this problem is very serious. 
Basically, all webpack projects cannot be migrated to esbuild, because esbuild does not support the separation of splitting chunk and common chunk.

I see a lot of such issues. I suggest that if it is not easy to do so, warn at compile time.

@evanw

**Comment by evanw** [comment]
I cannot reproduce your problem. Here's what I tried:

```
$ cat package.json
{
  "type": "module",
  "dependencies": {
    "esbuild": "0.12.25"
  }
}

$ cat build.js
import esbuild from 'esbuild'
esbuild.build({
  entryPoints: {
    a: './a.js',
    b: './b.js'
  },
  outdir: 'out',
})

$ cat a.js 
import { ref } from './c.js'
ref.value = 'hello'

$ cat b.js 
import { ref } from './c.js'
console.log(ref) // undefined, expect 'hello'

$ cat c.js 
export const ref = {}

$ node build.js

$ node out/b.js
node:internal/process/esm_loader:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module './out/c.js' imported from ./out/b.js
Did you mean to import ../../c.js?
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:389:11)
    at moduleResolve (node:internal/modules/esm/resolve:867:10)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:978:11)
    at Loader.resolve (node:internal/modules/esm/loader:89:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:242:28)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

The error about `c.js` is expected because you have only told esbuild to build `a.js` and `b.js`, not `c.js`. If you would like for esbuild to build and bundle dependencies too you would enable `bundle: true`. And if you want to enable code splitting, you would add `splitting: true, format: 'esm'`. Neither of these cases result in `ref` being undefined.

**Comment by yisar** [comment]
> And if you want to enable code splitting, you would add splitting: true, format: 'esm'

Yes, I want to do code splitting, but the output is not what I want. What I want is the packaged file, but the output of esbuild is an esm file containing `import` and `export`.

Please note that our code is not run in the browser, but executed through `eval` in JSCore and V8, it does not support native module.

This is my last obstacle to using esbuild in production. My leader said that if this problem can’t be resolved, esbuild can’t be used in a production environment anyway.

Is there any good idea here?

**Comment by hyrious** [comment]
@yisar
Maybe you could try the tsup way: bundle to esm with splitting enabled, and transform each result file to cjs.
BTW, [v8 does support es modules](https://v8.dev/features/modules).

**Comment by yisar** [comment]
> Maybe you could try the tsup way: bundle to esm with splitting enabled, and transform each result file to cjs.

Yes, you are right. In fact, I am using this method now, but it is not very good, because esbuild generates too many chunks and it does not support customization. I need to traverse the files and convert them (using swc), which will become slow.
